cmd/allocator/main.go

```diff
@@ -38,24 +46,71 @@ var (
 const (
 	certDir = "/home/allocator/client-ca/"
 	tlsDir  = "/home/allocator/tls/"
-	port    = "8443"
+	sslPort = "8443"
+
+	enableStackdriverMetricsFlag = "stackdriver-exporter"
+	enablePrometheusMetricsFlag  = "prometheus-exporter"
+	projectIDFlag                = "gcp-project-id"
 )
 
+func init() {
+	registerMetricViews()
+}
+
 // A handler for the web server
 type handler func(w http.ResponseWriter, r *http.Request)
 
 func main() {
+	conf := parseEnvFlags()
+
+	// Stackdriver metrics
+	if conf.Stackdriver {
+		sd, err := metrics.RegisterStackdriverExporter(conf.GCPProjectID)
+		if err != nil {
```

Comment:
can you move the code to a function e.g. setupMetricsRecorder(conf)?

Reply:
will do yes !